meson.build

```diff
@@ -0,0 +1,177 @@
+                         output: 'waltham-server.h',
+                         depend_files: 'tools/gen.py',
+                         command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'server', '-t', 'header'],
+                         install: true,
+                         install_dir: 'include/waltham')
+
+server_serialice = custom_target('server-serialice.c',
+                                 input: ['src/waltham/serial-preamble.txt', protocols_src],
+                                 output: 'server-serialice.c',
+                                 depend_files: 'tools/gen.py',
+                                 depends: server_h,
+                                 command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'server', '-t', 'marshaller'])
+
+server_deserialice = custom_target('server-deserialice.c',
+                                   input: ['src/waltham/deserial-preamble.txt', protocols_src],
+                                   output: 'server-deserialice.c',
+                                   depend_files: 'tools/gen.py',
+                                   depends: server_h,
+                                   command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'server', '-t', 'demarshaller'])
+
+generated_src = [
+    client_h,
+    client_serialice,
+    client_deserialice,
+    server_h,
+    server_serialice,
+    server_deserialice,
+]
+
+add_global_arguments([
+    '-Wall',
+    '-Wextra',
+    '-Wno-format-zero-length',
+    '-Wno-unused-parameter',
+    '-Wstrict-prototypes',
+    '-Wmissing-prototypes',
+    '-Wno-missing-field-initializers',
+    '-fvisibility=hidden',
+], language: 'c')
+
+libwaltham_inc = include_directories('src/waltham')
+libwaltham = shared_library('waltham', libwaltham_src, generated_src,
+                            include_directories: libwaltham_inc,
+                            link_args: '-Wl,--no-undefined',
+                            install: true,
+                            version: version,
+                            soversion: version[0])
+install_headers(libwaltham_headers, subdir: 'waltham')
+pkgconfig.generate(libraries: libwaltham,
+                   subdirs: 'waltham',
+                   version: version,
+                   name: 'Waltham',
+                   filebase: 'waltham',
+                   description: desc)
+
+tests_src = [
+    'tests/w-util.c',
+    'tests/w-util.h',
+]
+
+client_src = ['tests/client-api-example.c', client_h, tests_src]
+client_exe = executable('client', client_src, include_directories: libwaltham_inc, link_with: libwaltham)
+test('client', client_exe)
+
+server_src = ['tests/server-api-example.c', server_h, tests_src]
+server_exe = executable('server', server_src, include_directories: libwaltham_inc, link_with: libwaltham)
+test('server', server_exe)
```

Comment:
Neither `client` nor `server` should be declared as tests, because they are not automated tests.